openpgp/packet/private_key_test.go

```diff
@@ -56,6 +65,77 @@ func TestPrivateKeyRead(t *testing.T) {
 	}
 }
 
+func TestPrivateKeyEncrypt(t *testing.T) {
+	var encryptedMsg, decryptedMsg []byte
+	var encryptedC1, encryptedC2 *big.Int
+
+	for i, test := range privateKeyTests {
+		packet, err := Read(readerFromHex(test.privateKeyHex))
+		if err != nil {
+			t.Errorf("#%d: failed to parse: %s", i, err)
+			continue
+		}
+		privKey := packet.(*PrivateKey)
+		err = privKey.Decrypt([]byte("testing"))
+		if err != nil {
+			t.Errorf("#%d: failed to decrypt: %s", i, err)
+			continue
+		}
+
+		switch pubKey := privKey.PublicKey.PublicKey.(type) {
```

Comment:
Since the 2 changes I've made primarily affect `PrivateKey` encryption and serialization. Perhaps I should limit to testing those 2 things and leave the message encrypt-decrypt roundtrip to other tests. What do you think?